File: dastanlib/src/main/java/com/dastanapps/dastanlib/utils/LayoutUtils.java
package com.dastanapps.dastanlib.utils;

import android.support.constraint.ConstraintLayout;
import android.support.constraint.ConstraintSet;
import android.view.View;
import android.widget.RelativeLayout;

/**
 * Created by dastaniqbal on 19/07/2017.
 * 19/07/2017 6:30
 */

public class LayoutUtils {
    public static void setRLParams(int w, int h, int v, View rlView) {
        RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(w, h);
        params.addRule(v);
        rlView.setLayoutParams(params);
    }

    public static void setRLParams(int w, int h, View rlView, int... verb) {
        RelativeLayout.LayoutParams params = new RelativeLayout.LayoutParams(w, h);
        for (int aVerb : verb) {
            params.addRule(aVerb);
        }
        rlView.setLayoutParams(params);
    }

    public static void setConstraintRatio(ConstraintLayout layout, int viewId, String ratio) {
        ConstraintSet set = new ConstraintSet();
        set.clone(layout);
        set.setDimensionRatio(viewId, ratio);
        set.applyTo(layout);
    }
}

File: ui/src/main/java/com/dastanapps/dastanlib/utils/LayoutUtils.kt
package com.dastanapps.dastanlib.utils

import androidx.constraintlayout.widget.ConstraintLayout
import androidx.constraintlayout.widget.ConstraintSet
import android.view.View
import android.widget.RelativeLayout

/**
 * Created by dastaniqbal on 19/07/2017.
 * 19/07/2017 6:30
 */


object LayoutUtils {
    fun setRLParams(w: Int, h: Int, v: Int, rlView: View) {
        val params = RelativeLayout.LayoutParams(w, h)
        params.addRule(v)
        rlView.layoutParams = params
    }

    fun setRLParams(w: Int, h: Int, rlView: View, vararg verb: Int) {
        val params = RelativeLayout.LayoutParams(w, h)
        for (aVerb in verb) {
            params.addRule(aVerb)
        }
        rlView.layoutParams = params
    }

    fun setConstraintRatio(layout: ConstraintLayout, viewId: Int, ratio: String) {
        val set = ConstraintSet()
        set.clone(layout)
        set.setDimensionRatio(viewId, ratio)
        set.applyTo(layout)
    }
}
